node/pkg/processor/cleanup.go

```diff
@@ -73,20 +73,26 @@ func (p *Processor) handleCleanup(ctx context.Context) {
 			// This occurs when we observed a message after the cluster has already reached
 			// consensus on it, causing us to never achieve quorum.
 			if ourVaa, ok := s.ourObservation.(*VAA); ok {
-				if _, err := p.db.GetSignedVAABytes(*db.VaaIDFromVAA(&ourVaa.VAA)); err == nil {
-					// If we have a stored quorum VAA, we can safely expire the state.
-					//
-					// This is a rare case, and we can safely expire the state, since we
-					// have a quorum VAA.
-					p.logger.Info("Expiring late VAA", zap.String("digest", hash), zap.Duration("delta", delta))
-					aggregationStateLate.Inc()
-					delete(p.state.signatures, hash)
-					continue
-				} else if err != db.ErrVAANotFound {
-					p.logger.Error("failed to look up VAA in database",
-						zap.String("digest", hash),
-						zap.Error(err),
-					)
+				if ourVaa.VAA.EmitterChain == vaa.ChainIDPythNet {
```

Comment:
I feel like adding an emitter chain check every place we handle a vaa is going to become a maintenance hassle.  We'll have to remember to add this check for every new place that deals with concrete vaas.

I think extending the `Observation` interface is a less brittle way to do this.  Basically, we would define a `PythnetVAA` type and then implement the `Observation` interface for it.  This will automatically get us type safety because any place that downcasts to the `*VAA` type directly (like here on line 75) will not catch `PythnetVAA`s.  We would implement `HandleQuorum` by storing the vaa in the `pythnetVaas` map, which would guarantee at the type level that pythnet vaas don't end up in the db.

In general, any place where we're doing an emitter chain check for pythnet should be a method on the observation interface instead (other than the initial place where the struct is created).